### ISSUE TITLE
[HOLD for payment 2021-11-25] Update "Be the first person to comment!" screen when you start a new DM 

### ISSUE THREAD

**Issue by mallenexpensify** [Expensify/App#4678]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Sent DM message to someone I hadn't messaged before

## Expected Result:
The blank screen should have avatars, a message for "This is the beginning of your chat history with XXXX (pronouns), as well as showing the timezone above the message input when messaging someone 1:1. 

## Actual Result:
Only see "Be the first person to comment!"

## Solution
Let's update this state/area to include the users icon, name, preferred pronouns and the time in their time zone (only if 1:1 DM).  
![image](https://user-images.githubusercontent.com/2319350/130205221-8a4a27a3-84fc-46b6-a8ef-febfd0aa4f43.png)

Design files are [here in Figma ](https://www.figma.com/file/rReWxiEhGDHIjajsE9XoJw/Start-a-chat?node-id=0%3A1)

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Expensify/Expensify Issue URL:** [link](https://github.com/Expensify/Expensify/issues/142661)

[View all open jobs on Upwork](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Upwork job:** https://www.upwork.com/jobs/~0159561fdd76ccd036

**Comment by MelvinBot** [comment]
Triggered auto assignment to @michaelhaxhiu (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by Santhosh-Sellavel** [comment]
@mallenexpensify 
 
 What's it gonna be for the new group chat?  
 
<img width="461" alt="Screenshot 2021-08-17 at 12 37 42 AM" src="https://user-images.githubusercontent.com/85645967/129616732-97fdaa3e-ddb3-4136-804c-f856d1e21d4c.png">

**Comment by mallenexpensify** [comment]
Great point @Santhosh-Sellavel ! I addressed your question in an old internal issue we had been working on, I instructed folks to post comments/feedback to this issue.  We have two choices

1. Address group chats in this GH and figure out what we want before posting the job.  
2. Only focus on the single DM usecase for now and address groups later.

**Comment by Santhosh-Sellavel** [comment]
Is this going to be external? @mallenexpensify

**Comment by mallenexpensify** [comment]
It will very likely be external @Santhosh-Sellavel , @michaelhaxhiu needs to triage the issue first, part of that is figuring out what we want to do about group messages.  My hunch is it's best to fix this issue specifically for the 1:1 DM usecase now then create a new issue in the future to address both group messages and possibly rooms.  

@michaelhaxhiu, this shouldn't require a designer, an engineer should be able to take care of everything

**Comment by rosegrech** [comment]
Great question about the group. I want to say lets work on how the 1:1 looks like and then probably handle the group in a new GH to keep things neat and tidy. I'd love to say whatever we use for 1:1 will work for the group but ard to say that as of now :)

**Comment by shawnborton** [comment]
@megankelso @michelle-thompson I vaguely recall that we explored what this screen would look like for various use cases... actually I think it was @michelle-thompson that did it for Concierge Travel. 

So basically we could do something like this?
![image](https://user-images.githubusercontent.com/2319350/129685211-c82ec633-a746-4527-9666-7102a7c18074.png)


And maybe before we have those features ready to go, we could just use a generic message for the group situation like so:
![image](https://user-images.githubusercontent.com/2319350/129685336-af524d5e-19c5-43f2-badf-26e602874053.png)

And then in terms of scalability, looks like Michelle had mocked up something like this for when a group has lots of participants:
![image](https://user-images.githubusercontent.com/2319350/129685468-6160e3c4-d32c-4759-8a06-9bcd359bd478.png)

**Comment by michelle-thompson** [comment]
Cool, happy to move forward with the generic messaging you've mocked up above!

**Comment by shawnborton** [comment]
Sweet, so I think maybe we want to make the avatars in my mockup a bit smaller like your original mocks (the one at the bottom) for scalability purposes. Otherwise, then I think we just need to decide if we're going to add Group chats to the exported Upwork issue?

**Comment by michelle-thompson** [comment]
Ahh, you mean so the size of the avatar never has to change? And I figure might as well add group chats to this issue.

**Comment by shawnborton** [comment]
Yeah, I think I like the size you had here:
![image](https://user-images.githubusercontent.com/2319350/129773051-6dbb1991-41e0-4cd8-830c-4868db745c5a.png)

...better than the larger size I used. So that gives us something like this:
![image](https://user-images.githubusercontent.com/2319350/129773863-88f558ec-b350-4fda-af92-b1ae64ed87c7.png)

**Comment by michelle-thompson** [comment]
Agreed, I think that looks good too

**Comment by megankelso** [comment]
I agree, this looks good! I do wonder if we should make this a bit more consistent. What comes to mind for me is using the existing UI pattern to show local time (above the chat input) and including the same "this is the beginning for your chat" message, but perhaps adding the pronouns. Curious for thoughts?

<img width="968" alt="Screen Shot 2021-08-17 at 11 19 20 AM" src="https://user-images.githubusercontent.com/49413768/129779630-07cbd313-a51c-4dee-9835-c553c18d79cd.png">

**Comment by michelle-thompson** [comment]
I'm into it! Curious what this would look like if you were to DM someone in the same time zone?

**Comment by megankelso** [comment]
I'm suggesting it still show the time for the first chat, regardless of what time zone. Curious if you had something else in mind?

**Comment by michelle-thompson** [comment]
Ahhh, I see. I think that would be fine too, I'm keen with either direction!

**Comment by shawnborton** [comment]
Oh, I really like that idea about the timezone. Actually taking this a step further, perhaps the name above the message for the 1:1 case is redundant as well? So that leaves us with this:

![image](https://user-images.githubusercontent.com/2319350/129869799-2fc3be5d-cee0-4011-af46-ba94ff341881.png)

**Comment by megankelso** [comment]
yes, agreed! we can drop the name. I like the addition of pronouns to the group but I'm not sure how scaleable it is for larger group chats. For example, if we added them to the next screen:

<img width="316" alt="Screen Shot 2021-08-18 at 11 48 15 AM" src="https://user-images.githubusercontent.com/49413768/129955068-cd606405-63eb-4720-8894-f3c27f80ae20.png">

Curious for thoughts. would it be weird to make a rule that pronouns are included only for group chats of 3 people and under or so?

**Comment by michelle-thompson** [comment]
I think you can have up to eight people in DMs, so it's possible to include pronouns for everyone! It looks a little busy, but I don't particularly mind that since this screen is temporary.

![image](https://user-images.githubusercontent.com/43361905/129974739-450c5599-c1bb-4ae8-891c-4e11df9eee82.png)

**Comment by shawnborton** [comment]
I agree, I think it will be okay. It's probably not very likely that all 8 members of the group will have their pronouns set anyways.

**Comment by MelvinBot** [comment]
@michaelhaxhiu Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by mallenexpensify** [comment]
@shawnborton @michelle-thompson will let you @michaelhaxhiu know when he should review it for triage? Also.. huge, double plus, bonus points if you can update the OP with the deliverables for the contributors.

**Comment by shawnborton** [comment]
Cool, I updated the original comment with the new screenshots, I think this is ready for @michaelhaxhiu

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by michaelhaxhiu** [comment]
Posted to upwork just now, and added job link in GH body.

**Comment by MirFahad58** [comment]
<img width="1292" alt="Screenshot 2021-08-24 at 12 48 30 AM" src="https://user-images.githubusercontent.com/31244700/130510117-adc7d397-5a3b-4b70-964e-fa1f60b7e8f9.png">

As shown in the picture, I suggest we can fix this by placing it above the text input not at the centre in a fixed position. 
reasons: 
1 - on smaller screens fixed positions (which is the centre) will be under the keyboard. so it will not be visible. above the keyboard, we can use keyboard avoiding view and will be in the best position, not mater user have a smaller or larger device
2 - by default keyboard should be open (auto-focused) because the user is here to start the chat. this will  reduce one more step
3 - In a group chat, we should see only two in that group and with others tag if he clicks other we can show other people in that chat as details, otherwise the user could be overwhelmed.
4 - I recommend "write a message" over "say hello" because it is not a social app or fun app, a professional app should use professional words.

**Comment by shawnborton** [comment]
I like the feedback that the keyboard should be open by default on mobile devices. However, let's please stick with our mockups where the empty state is vertically centered in the available empty space. We also want to keep "Say hello" as it's in line with our brand tone.

**Comment by michaelhaxhiu** [comment]
That's a good start @MirFahad58, nice!

I agree with Shawn's feedback regarding sticking to vertical centering, as shown in the mockups. 

>I like the feedback that the keyboard should be open by default on mobile devices.

Good call, one less step for the user. Can you show a quick screenshot of this screen with the keyboard expanded?

**Comment by MirFahad58** [comment]
Ok

**Comment by michaelhaxhiu** [comment]
Sounds good. @puneetlath is assigned to review proposals here.

**Comment by MirFahad58** [comment]
this one is with sticking to vertical centring.
<img width="432" alt="Screenshot 2021-08-24 at 7 54 25 PM" src="https://user-images.githubusercontent.com/31244700/130639827-41eef198-bcc6-4b4e-b93c-00a78bff2d93.png">

and this one is for what I was suggesting.
<img width="458" alt="Screenshot 2021-08-24 at 7 53 32 PM" src="https://user-images.githubusercontent.com/31244700/130639919-60417e3e-1f81-44a2-a44f-8bb0057ba03c.png">

**Comment by shawnborton** [comment]
Please stick with the style from our mockups. It looks like your mockups have a different font as well as the timezone being centered.

**Comment by MirFahad58** [comment]
yeah, I just wanted to share the idea, just created my separate mock to get ideas and work around it.

**Comment by MirFahad58** [comment]
the development will be based on styles from your mockups.

**Comment by puneetlath** [comment]
@michaelhaxhiu FYI I'm focused on allocations and then OOO the rest of the week, so going to unassign myself.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @roryabraham (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by roryabraham** [comment]
Okay, despite not making a formal proposal, it seems like @MirFahad58 has a pretty good grip on this task. @MirFahad58 If you're interested in working on this, can you please write up a proposal for how you would implement this in the code?

**Comment by MirFahad58** [comment]
yeah sure @roryabraham

**Comment by MirFahad58** [comment]
First, I will understand the flow in the app, and what you guys had used (like a UI library) or used custom components, so text inputs and other components must be already there, I will reuse with my purpose and styling, And again I will look in the code structure, how flexes are implemented.. etc
I will start from the build a basic structure of the screen by adding flexes for each section, after that the header part and make a separate one so we can reuse it for different chats like group chats etc.
After that will create a custom avatar as a separate component so we can pass the source and height width as props;
Conditionally if there is no chat history we will show the "be the first person to comment" screen. and Input will have autoFocus so the keyboard should auto open.
but for all this, I have to go through the code base and see which components are prebuilt and I can reuse those.

**Comment by roryabraham** [comment]
Okay, good enough for me @MirFahad58. @michaelhaxhiu let's hire @MirFahad58 on Upwork.

@MirFahad58 Since I believe this will be your first pull request in this repo, let me give you some pointers to help you succeed and to help ensure that the PR review moves quickly. I strongly recommend that you read over these resources before submitting your PR and refer to them as needed while you work:

1. Please read our [JavaScript style guide](https://github.com/Expensify/App/blob/main/STYLE.md).
    - If necessary, prefer the use of [Underscore.js](https://underscorejs.org/) over built-in JavaScript array methods. Underscore has lots of useful utilities and generally has better null-checking.
    - Also use `lodashGet` instead of any optional chaining.
1. Read over the [style guide](https://github.com/Expensify/App/blob/main/STYLING.md). This provides guidance on creating styles in this repo. We want to keep our styles as generic and reusable as possible.
1. Read over the [structure of the app](https://github.com/Expensify/App#structure-of-the-app). In particular, the [directory structure](https://github.com/Expensify/App#directory-structure), [file naming/structure](https://github.com/Expensify/App#file-namingstructure), and [platform-specific style extensions](https://github.com/Expensify/App#platform-specific-file-extensions) will be relevant in this PR. We do not use the react-native `Platform` library (i.e: no `Platform.select`).
1. Read the [contributing guidelines](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md).

Thanks! If you have any questions, refer to ["asking questions"](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#asking-questions) in the contributing guidelines.

**Comment by MirFahad58** [comment]
Ok thanks

**Comment by shawnborton** [comment]
In terms of building the empty state component, keep in mind that this component will be reused for rooms as well. So we want to make sure we can extend this component to support a custom room avatar and a custom message below the avatar as well:
![image](https://user-images.githubusercontent.com/2319350/130749881-5bce328c-fd35-4114-827e-921b4939d4a2.png)

**Comment by MirFahad58** [comment]
Yeah sure

**Comment by michaelhaxhiu** [comment]
Hired on upwork!

**Comment by MirFahad58** [comment]
Started working on the issue, linted the code, but was unable to commit. need someone to help.

**Comment by MirFahad58** [comment]
<img width="432" alt="Screenshot 2021-08-28 at 11 00 55 PM" src="https://user-images.githubusercontent.com/31244700/131226791-5d7f54f7-22f0-4c17-8ee2-70f27224856b.png">
getting verified tag for commits on other repos but unable to commit on this one, getting gpg error and already done with gpg step by step process.

**Comment by Santhosh-Sellavel** [comment]
@MirFahad58
> commit on this one?

Are you trying to push your changes directly in this repo

**Comment by MirFahad58** [comment]
no, forked and created a new branch.

**Comment by MirFahad58** [comment]
<img width="381" alt="Screenshot 2021-08-28 at 11 32 53 PM" src="https://user-images.githubusercontent.com/31244700/131227562-2189bb0d-3035-4e5e-ae31-2736f1a00903.png">
getting this error

**Comment by Santhosh-Sellavel** [comment]
Are you using the GUI to commit?
If then, check whether you configured git settings correctly.

Try in the terminal using the following command, this works like charm!
> git commit -S -m "Your Commit message"

**Comment by MirFahad58** [comment]
@Santhosh-Sellavel  Same for both

**Comment by Santhosh-Sellavel** [comment]
Try going through this documentation, This will help.
https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/signing-commits

Clone the repo in a new folder, try to set up this first! @MirFahad58

**Comment by MirFahad58** [comment]
resolved, thanks a lot @Santhosh-Sellavel

**Comment by MirFahad58** [comment]
I need to know, do I have to just create the screen or have to integrate with data also?
I am not clear.

**Comment by MirFahad58** [comment]
@michaelhaxhiu @roryabraham done with the issue, here are the screenshots before creating PR I am improving the code. but here is the 1st look.
<img width="428" alt="Screenshot 2021-08-29 at 5 27 51 PM" src="https://user-images.githubusercontent.com/31244700/131250827-4db3b2e0-1444-4ded-a399-74cde90c3903.png">
<img width="444" alt="Screenshot 2021-08-29 at 5 29 05 PM" src="https://user-images.githubusercontent.com/31244700/131250831-8953fa65-c456-48d7-b49c-475938892386.png">
<img width="1920" alt="Screenshot 2021-08-29 at 5 29 22 PM" src="https://user-images.githubusercontent.com/31244700/131250832-49042f5a-f498-4b9d-87bf-7dcf7d2869a6.png">
<img width="1915" alt="Screenshot 2021-08-29 at 5 29 44 PM" src="https://user-images.githubusercontent.com/31244700/131250833-f8076ef8-b3c8-42d7-9aed-fc4baa181978.png">

**Comment by MirFahad58** [comment]
<img width="497" alt="Screenshot 2021-08-29 at 6 17 00 PM" src="https://user-images.githubusercontent.com/31244700/131251720-f000cbed-4159-4303-9834-d101a12d7a01.png">

all test cases are passed.

**Comment by shawnborton** [comment]
Hi @MirFahad58 - it looks like you are not using the correct style for the timezone. Please reference the mockups for the correct styles, but we should already have these styles in place as we currently show the timezone for a user when you message someone outside of your own timezone. 

Could you also please put a space ` ` between the Name and (pronouns) as well? Thanks!

**Comment by MirFahad58** [comment]
Yeah already done screenshot were little bit old

**Comment by shawnborton** [comment]
Sounds good, we look forward to seeing your PR!

**Comment by MirFahad58** [comment]
@shawnborton pull request should be to "main" branch?

**Comment by shawnborton** [comment]
@MirFahad58 yup, that is correct.

**Comment by MirFahad58** [comment]
need help how to sign CLA

**Comment by mallenexpensify** [comment]
> CLA - You must sign our Contributor License Agreement by following the CLA bot instructions that will be posted on your PR
https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#user-content-submit-your-pull-request-for-final-review

Are you still having issues @MirFahad58 ?

**Comment by MirFahad58** [comment]
No thanks

**Comment by michaelhaxhiu** [comment]
As an update:
- @MirFahad58 has updated his PR and we are waiting on @shawnborton to review. 
- I inquired with @roryabraham in New Expensify to see if he may be able to review it sooner. 
- If not, let's wait on Shawn to review monday/tuesday of next week.

@MirFahad58 you had a few questions in Upwork Chat about payment earlier today, so I wanted to refer you to the [Payment for Contributors](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#payment-for-contributions) section of the **[Contributing to Expensify](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md)** guide. If you have follow up questions, let's discuss on this GH issue per the instructions. Thanks!

**Comment by MirFahad58** [comment]
Ok, I got it no more questions.

**Comment by michaelhaxhiu** [comment]
Update:
- @MirFahad58 had some personal commitments last week, and was able to update the PR for review https://github.com/Expensify/App/pull/4921#issuecomment-917553733
- Next up, we are waiting on Rory to review

**Comment by kadiealexander** [comment]
Hi @MirFahad58, I've placed this issue on hold as per [this update](https://expensify.slack.com/archives/C01GTK53T8Q/p1631754482374800), as we are prioritising issues related to a feature release scheduled for 9/24. As an apology for the delay, we will add a $100 bonus as a thank you for waiting.

**Comment by MirFahad58** [comment]
@kadiealexander  ok thanks i will wait for that.

**Comment by michaelhaxhiu** [comment]
I added the $100 as a milestone to the Upwork job.

**Comment by kadiealexander** [comment]
@MirFahad58 to clarify, you can feel free to work on this and submit it for review, we just are unable to merge any changes until after 9/24. 😊

**Comment by michaelhaxhiu** [comment]
@MirFahad58 the old upwork job expired so I re-posted a new one here. Can you apply so I can re-hire you? I added the $100 additional so the new job total is $350.

https://www.upwork.com/jobs/~01225edf8b252ecf69

**Comment by MirFahad58** [comment]
Ok . Asap i will do

**Comment by MirFahad58** [comment]
@michaelhaxhiu it isn't expired yet I think still in the active section. meanwhile, I submitted proposal on new one also.

**Comment by michaelhaxhiu** [comment]
Ah you are correct @MirFahad58, thanks -- I will close the new one sorry for the confusion.

**Comment by MirFahad58** [comment]
thanks

**Comment by parasharrajat** [comment]
I have started to review the PR and I will let you know @roryabraham when this is ready for review.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jboniface (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MirFahad58** [comment]
Hello @michaelhaxhiu @shawnborton i need the translation for these in es.js
reportActionsView: {
        beginingOfChatHistroy: 'This is the beginning of your chat history with',
        beginingOfChatHistroyPrivate: 'This is the beginning of the private',
        beginingOfChatHistroyPrivateSectionPart: 'room, invite others by @mentioning them.',
    },

**Comment by MirFahad58** [comment]
let me try this

**Comment by michaelhaxhiu** [comment]
@MirFahad58 here's the spanish translation - let us know if you are blocked on anything else:

>Este es el comienzo de la sala privada ${name}, invita a más gente @mencionándolos

**Comment by parasharrajat** [comment]
Guys, PR is ready for final review. These changes are were long back. :smile:

**Comment by MirFahad58** [comment]
we did that change long time ago.

**Comment by MirFahad58** [comment]
@michaelhaxhiu @parasharrajat @shawnborton @roryabraham no one is reviewing the PR?

**Comment by michaelhaxhiu** [comment]
Oh I see. I am seeing that the PR has the `n6-hold` label. Let me double check if this PR review needs to be held longer or not.

**Comment by jboniface** [comment]
not sure where this one is at

**Comment by roryabraham** [comment]
I left some review comments about a week ago but the PR hasn't been updated. I know there have been lots of delays on our side, but @MirFahad58 let's get this issue over the finish line!

**Comment by MirFahad58** [comment]
@roryabraham i will update asap.

**Comment by MirFahad58** [comment]
@michaelhaxhiu @roryabraham  @shawnborton 
I think PR is ready to merge all checks have passed.
<img width="1157" alt="Screenshot 2021-11-17 at 1 40 51 PM" src="https://user-images.githubusercontent.com/31244700/142166583-3292f470-9930-4913-8fed-a44d27c14024.png">

**Comment by MirFahad58** [comment]
and I left some messages on U.W need the response

**Comment by michaelhaxhiu** [comment]
@MirFahad58 can you make sure to send any messages concerning your progress in GH? We have more visibility here and there are more eyes to weigh in on your inquiries.

**Comment by MirFahad58** [comment]
@michaelhaxhiu I requested a bonus milestone for n6-hold on U.W and want to submit the work for the current milestone as I have to match my expenses for this month. Thanks

**Comment by roryabraham** [comment]
PR merged 🎉

**Comment by MirFahad58** [comment]
@michaelhaxhiu  as PR is merged now. I need response for above comment. Please reply

**Comment by michaelhaxhiu** [comment]
@MirFahad58 

We'll pay the remaining amount due after 7 days of no regressions. 

Let's do a quick buddy check on the outstanding payment that's owed for your work, as I'd like to make sure it's fair!

1. $500 for completing the job (The job was initially $250, but we increased that to $500 due to increase scope - [context](https://github.com/Expensify/App/issues/4678#issuecomment-944268717))
2. $250 bonus for the n6 hold (Already paid via Milestone in Upwork)

![image](https://user-images.githubusercontent.com/13206896/142472835-52fb45fc-2ea9-48de-9568-f6719daede70.png)

**Comment by MirFahad58** [comment]
@michaelhaxhiu let men correct, the 250 that was paid,  was not of n6-hold bonus it was half of $500 from jobs. Thanks
that's why i added a milestone for n6-hold

<img width="763" alt="Screenshot 2021-11-18 at 11 21 01 PM" src="https://user-images.githubusercontent.com/31244700/142473986-65cb5ba4-3272-41c6-82a4-f6a0a90f8159.png">

**Comment by michaelhaxhiu** [comment]
Ok so it sounds like you are saying:

1. The $250 that was already paid was for your initial work ✅ 
2. The $250 n6 hold bonus hasn't been sent yet, and you are requesting that be sent today 
3. Finally, we owe $250 for the increased scope of work, which will be paid in 6 days. 

Is that right?

**Comment by MirFahad58** [comment]
right

**Comment by michaelhaxhiu** [comment]
Ok got it. I think that is fine, I'll issue the $250 payment now.

Then, you'll receive the remaining $250 on November 25.

**Comment by MirFahad58** [comment]
yup thanks

**Comment by mallenexpensify** [comment]
Paid final $250 to @MirFahad58 , thanks for the help and patience.

**Comment by botify** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.1.16-10 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/4921


If no regressions arise, payment will be issued on 2021-12-06. :confetti_ball:
